File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/PlayerProfile.java
package pize.tests.voxelgame.main.net;

import pize.math.Maths;
import pize.util.StringUtils;

public class PlayerProfile{
    
    private final String name;
    
    public PlayerProfile(String name){
        this.name = name;
    }
    
    
    public String getName(){
        return name;
    }
    
    public static boolean isNameInvalid(String name){
        return StringUtils.isBlank(name) || name.length() > 16 || name.length() < 3 || name.contains(" ");
    }
    
    public static String genFunnyName(){
        final String[] funnyNames = {"Makcum", "Kriper", "IlyaPro", "ViktorPlay", "Kirbo", "IbremMiner", "intbyte", "ABelevka", "Dmitry"};
        return funnyNames[Maths.random(0, funnyNames.length - 1)] + Maths.random(51, 99);
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/PlayerProfile.kt
package pize.tests.voxelgame.main.net

import pize.math.Maths.random
import pize.util.StringUtils.isBlank

class PlayerProfile(val name: String?) {

    companion object {
        fun isNameInvalid(name: String?): Boolean {
            return isBlank(name) || name!!.length > 16 || name.length < 3 || name.contains(" ")
        }

        fun genFunnyName(): String {
            val funnyNames = arrayOf(
                "Makcum",
                "Kriper",
                "IlyaPro",
                "ViktorPlay",
                "Kirbo",
                "IbremMiner",
                "intbyte",
                "ABelevka",
                "Dmitry"
            )
            return funnyNames[random(0, funnyNames.size - 1)] + random(51, 99)
        }
    }
}
